pkg/kube/snapshot/snapshot_alpha.go

```diff
@@ -0,0 +1,317 @@
+	VolSnapClassGVR = schema.GroupVersionResource{Group: v1alpha1.GroupName, Version: v1alpha1.Version, Resource: v1alpha1.VolumeSnapshotClassResourcePlural}
+	// VolSnapContentGVR specifies GVR schema for VolumeSnapshotContents
+	VolSnapContentGVR = schema.GroupVersionResource{Group: v1alpha1.GroupName, Version: v1alpha1.Version, Resource: v1alpha1.VolumeSnapshotContentResourcePlural}
+)
+
+type SnapshotAlpha struct {
+	kubeCli kubernetes.Interface
+	dynCli  dynamic.Interface
+}
+
+func (sna *SnapshotAlpha) GetVolumeSnapshotClass(annotationKey, annotationValue string) (string, error) {
+	us, err := sna.dynCli.Resource(VolSnapClassGVR).Namespace("").List(metav1.ListOptions{})
+	if err != nil {
+		return "", errors.Errorf("Failed to get VolumeSnapshotClasses in the cluster: %v", err)
+	}
+	if us == nil || len(us.Items) == 0 {
+		return "", errors.Errorf("Failed to find any VolumeSnapshotClass in the cluster: %v", err)
+	}
+	for _, vsc := range us.Items {
+		ans := vsc.GetAnnotations()
+		if val, ok := ans[annotationKey]; ok && val == annotationValue {
+			return vsc.GetName(), nil
+		}
+	}
+	return "", errors.Errorf("Failed to find VolumesnapshotClass with %s annotation in the cluster", annotationKey)
+}
+
+// Create creates a VolumeSnapshot and returns it or any error happened meanwhile.
+func (sna *SnapshotAlpha) Create(ctx context.Context, name, namespace, volumeName string, snapshotClass *string, waitForReady bool) error {
+	if _, err := sna.kubeCli.CoreV1().PersistentVolumeClaims(namespace).Get(volumeName, metav1.GetOptions{}); err != nil {
+		if k8errors.IsNotFound(err) {
+			return errors.Errorf("Failed to find PVC %s, Namespace %s", volumeName, namespace)
+		}
+		return errors.Errorf("Failed to query PVC %s, Namespace %s: %v", volumeName, namespace, err)
+	}
+
+	err := sna.createVolumeSnapshot(name, namespace, corev1.ObjectReference{Kind: pvcKind, Name: volumeName}, *snapshotClass)
+	if err != nil {
+		return err
+	}
+
+	if !waitForReady {
+		return nil
+	}
+
+	err = sna.WaitOnReadyToUse(ctx, name, namespace)
+	if err != nil {
+		return err
+	}
+
+	_, err = sna.Get(ctx, name, namespace)
+	return err
+}
+
+// Get will return the VolumeSnapshot in the namespace 'namespace' with given 'name'.
+func (sna *SnapshotAlpha) Get(ctx context.Context, name, namespace string) (*v1alpha1.VolumeSnapshot, error) {
+	us, err := sna.dynCli.Resource(VolSnapGVR).Namespace(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	vs := &v1alpha1.VolumeSnapshot{}
+	err = TransformUnstructured(us, vs)
+	return vs, err
+}
+
+// Delete will delete the VolumeSnapshot and returns any error as a result.
+func (sna *SnapshotAlpha) Delete(ctx context.Context, name, namespace string) error {
+	if err := sna.dynCli.Resource(VolSnapGVR).Namespace(namespace).Delete(name, &metav1.DeleteOptions{}); !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	// If the Snapshot does not exist, that's an acceptable error and we ignore it
+	return nil
```

Comment:
While I agree this is correct, maybe from a readability standpoint it would be better if 
```suggestion
	if err := sna.dynCli.Resource(VolSnapGVR).Namespace(namespace).Delete(name, &metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
		return err
	}
	return nil
```